### ISSUE TITLE
Error in CMakeLists for flute-ffpmeg

### ISSUE THREAD

**Issue by jordijoangimenez** [5G-MAG/rt-mbms-examples#10]
I'm getting the following error:

```
fivegmag@5GMAG-NUC:~$ cd rt-mbms-examples/flute-ffmpeg
mkdir build && cd build
cmake -GNinja ..
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found Boost: /usr/lib/x86_64-linux-gnu/cmake/Boost-1.74.0/BoostConfig.cmake (found version "1.74.0")
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libpistache'
--   Found libpistache, version 0.3.1-git20240626
CMake Error at lib/CMakeLists.txt:1 (add_subdirectory):
  add_subdirectory given source "rt-libflute" which is not an existing
  directory.

```

should rt-libflute be copied inside lib?

**Comment by dsilhavy** [comment]
This should be covered by `.gitmodules`: https://github.com/5G-MAG/rt-mbms-examples/blob/development/flute-ffmpeg/.gitmodules

I assume we need to move the content of  `flute-ffmpeg/.gitmodules` to `.gitmodules` in the root folder of the project and then link all dependencies of the subprojects. Otherwise, the dependencies are missing for the respective subprojects if user clone the whole repo. 

For now you should be able to fix this by simply cloning `rt-libflute` into the `lib` folder

**Comment by jordijoangimenez** [comment]
We could fix:

## Build

### Clone the repository

````
git clone --recurse-submodules https://github.com/5G-MAG/rt-mbms-examples
````

### Clone the rt-libflute repository inside the lib folder

````
cd rt-mbms-examples/lib
git clone --recurse-submodules https://github.com/5G-MAG/rt-libflute
```` 

### Build setup

````
cd rt-mbms-examples/flute-ffmpeg
mkdir build && cd build
cmake -GNinja ..
````

**Comment by dsilhavy** [comment]
Thanks @jordijoangimenez I think the fix with adding the dependency to `.gitmodules` is a bit cleaner. I fixed this in #11 and https://github.com/5G-MAG/rt-mbms-examples/commit/ff89b08ac7dcd70c184cb3850c006ac214daa71c.
